test/e2e/serial/tests/configuration.go

```diff
@@ -345,5 +350,196 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 		})
+
+		It("[test_id:47585][reboot_required][slow] can change kubeletconfig and controller should adapt", func() {
+			nroOperObj := &nropv1alpha1.NUMAResourcesOperator{}
+			err := fxt.Client.Get(context.TODO(), client.ObjectKey{Name: objects.NROName()}, nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", objects.NROName())
+
+			initialNrtList := nrtv1alpha1.NodeResourceTopologyList{}
+			initialNrtList, err = e2enrt.GetUpdated(fxt.Client, initialNrtList, timeout)
+			Expect(err).ToNot(HaveOccurred(), "cannot get any NodeResourceTopology object from the cluster")
+
+			mcps, err := nropmcp.GetNodeGroupsMCPs(context.TODO(), fxt.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred(), "cannot get MCPs associated with NUMAResourcesOperator %q", nroOperObj.Name)
+
+			kcList := &machineconfigv1.KubeletConfigList{}
+			err = fxt.Client.List(context.TODO(), kcList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var targetedKC *machineconfigv1.KubeletConfig
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						// pick the first one you find
+						targetedKC = kc
+					}
+				}
+			}
+			Expect(targetedKC).ToNot(BeNil(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			By("modifying reserved CPUs under kubeletconfig")
+			kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+
+			initialRsvCPUs := kcObj.ReservedSystemCPUs
+			applyNewReservedSystemCPUsValue(&kcObj.ReservedSystemCPUs)
+			err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = fxt.Client.Update(context.TODO(), targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for MachineConfigPools to get updated")
+			var wg sync.WaitGroup
+			for _, mcp := range mcps {
+				wg.Add(1)
+				go func(mcpool *machineconfigv1.MachineConfigPool) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					err = e2ewait.ForMachineConfigPoolCondition(fxt.Client, mcpool, machineconfigv1.MachineConfigPoolUpdated, configuration.MachineConfigPoolUpdateInterval, configuration.MachineConfigPoolUpdateTimeout)
+					Expect(err).ToNot(HaveOccurred())
+				}(mcp)
+			}
+			wg.Wait()
+
+			By("checking that NUMAResourcesOperator's ConfigMap has changed")
+			cmList := &corev1.ConfigMapList{}
+			err = fxt.Client.List(context.TODO(), cmList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var nropCm *corev1.ConfigMap
+			for i := 0; i < len(cmList.Items); i++ {
+				if objects.IsOwnedBy(cmList.Items[i].ObjectMeta, nroOperObj.ObjectMeta) {
```

Comment:
this likely means we will need to report the `configMap` namespace + name in the operator status, like we do for deployment. Good enough for now, though.